src/main/java/org/aksw/simba/squirrel/metadata/CrawlingActivity.java

```diff
@@ -139,17 +220,41 @@ public Worker getWorker() {
         return worker;
     }
 
-    public enum CrawlingURIState {SUCCESSFUL, UNKNOWN, FAILED;}
+    public enum CrawlingURIState {SUCCESSFUL, UNKNOWN, FAILED}
 
     public CrawlingURIState getState() {
         return state;
     }
 
-    public CrawleableUri getCrawleableUri() {
-        return uri;
+    public URI getUrl (String o)
+    {
+            try {
+                URL Domain = new URL("http://www.example.org/");
+                URL url = new URL(Domain + o);
+
+                return getUri(url);
+
+            } catch (MalformedURLException e) {
+                 LOGGER.info("MalformedURLException" + e);
+                e.printStackTrace();
+                 return null;
+
+            }
     }
 
-    public String getGraphId() {
-        return graphId;
+    public URI getUri(URL u)
```

Comment:
Why is this class taking a URL object as parameter, when it is simply using it as a String?